### ISSUE TITLE
CSS File show color hint instantly

### ISSUE THREAD

**Issue by pebbles186** [eclipse/wildwebdeveloper#285]
Steps to reproduce:
1. Open a .css file with Generic Text Editor
2. Add some code with a color, e.g.:
`.yellow-class {
  color: \#fdc400
}`

Expected:
A small square with the color is shown. (see picture)
![class_with_colored_attribute](https://user-images.githubusercontent.com/33489439/66894264-41677700-eff0-11e9-875a-985d6de60b69.png)

What actually happens
Nothing is displayed, until I open the "CSS editor".
Then in both files I can see the color.

The problem also exists the other way around. So opening CSS Editor first and then Generic Text Editor.

Some details about my installation:
Eclipse IDE for PHP Developers
Version: 2019-09 R (4.13.0)
Build id: 20190917-1200
Wild Web Developer 0.7.0.201910102046

**Comment by mickaelistria** [comment]
I can't reproduce this, the color preview appears promptly here.

**Comment by pebbles186** [comment]
Okay, on my side I can reproduce it. Maybe it is because I am on windows...
Are there any log files I can check? In the eclipse error log are no errors.

**Comment by angelozerr** [comment]
@mickaelistria I noticed this bug sometimes and I think it's an issue with LSP4E. You have the same trouble with folding. Generally when color is not displayed, folding is not done too.

The problem occurs when a `textDocument/didOpen `is done AFTER the `textDocument/foldingRange` and `textDocument/documentColor` LSP request. In this case, the CSS language server has not registered the document and disable those folding and documentColor features (perhaps teh CSS language server throws this error https://github.com/vscode-langservers/vscode-css-languageserver/blob/master/src/cssServerMain.ts#L375 and in this case it disable Eclipse CodeMining?).

In other words, LSP4E should take care of that `textDocument/didOpen ` requets should be donne before other requests. I'm afraid that it's an hard issue.

**Comment by mickaelistria** [comment]
> In other words, LSP4E should take care of that textDocument/didOpen requets
> should be donne before other requests. I'm afraid that it's an hard issue.


OK. Do you know if some ticket is already open against LSP4E on that matter?

**Comment by angelozerr** [comment]
> OK. Do you know if some ticket is already open against LSP4E on that matter?

No sorry I have not taken time to write it.

**Comment by mickaelistria** [comment]
> I'm afraid that it's an hard issue.

Actually, I don't think it is ;) However I'm not available to write on it very soon (I've got a bunch of topics to close first), but if anyone is interested, the key seems to be that the `LanguageServerWrapper.connect(...)` methods return a future and that the consumer of this method wait for the future completion before chaining operations.

**Comment by mickaelistria** [comment]
I couldn't resist... https://git.eclipse.org/r/151166

**Comment by mickaelistria** [comment]
> No sorry I have not taken time to write it.

Actually, you already did ;) https://bugs.eclipse.org/bugs/show_bug.cgi?id=548610

**Comment by pebbles186** [comment]
@mickaelistria and @angelozerr many thanks for your help :)

**Comment by angelozerr** [comment]
> Actually, you already did ;) https://bugs.eclipse.org/bugs/show_bug.cgi?id=548610

Wow don't remember that I have written that :)

**Comment by mickaelistria** [comment]
Please try upgrading to http://download.eclipse.org/lsp4e/snapshots/ where the issue should be fixed. If it doesn't help, feel free to reopen.
This should be fixed in future release of LSP4E+Wild Web Developer, end of December for the latest, but could happen earlier...
